lib/matplotlib/axes/_axes.py

```diff
@@ -3713,46 +3710,47 @@ def get_next_color():
                     shadow_dict.update(shadow)
                 self.add_patch(mpatches.Shadow(w, **shadow_dict))
 
-            if labeldistance is not None:
-                xt = x + labeldistance * radius * math.cos(thetam)
-                yt = y + labeldistance * radius * math.sin(thetam)
-                label_alignment_h = 'left' if xt > 0 else 'right'
-                label_alignment_v = 'center'
-                label_rotation = 'horizontal'
-                if rotatelabels:
-                    label_alignment_v = 'bottom' if yt > 0 else 'top'
-                    label_rotation = (np.rad2deg(thetam)
-                                      + (0 if xt > 0 else 180))
-                t = self.text(xt, yt, label,
-                              clip_on=False,
-                              horizontalalignment=label_alignment_h,
-                              verticalalignment=label_alignment_v,
-                              rotation=label_rotation,
-                              size=mpl.rcParams['xtick.labelsize'])
-                t.set(**textprops)
-                texts.append(t)
-
-            if autopct is not None:
-                xt = x + pctdistance * radius * math.cos(thetam)
-                yt = y + pctdistance * radius * math.sin(thetam)
+            theta1 = theta2
+
+        pc = PieContainer(slices, x, normalize)
+
+        if labels is not None and labeldistance is not None:
+            # Add labels to the wedges.
+            default_size = {'fontsize': mpl.rcParams['xtick.labelsize']}
+            if textprops is None:
+                labels_textprops = default_size
+            elif 'fontsize' in textprops or 'size' in textprops:
+                labels_textprops = textprops
+            else:
+               labels_textprops = textprops | default_size
```

Comment:
This is correct, but can we be shorter/clearer?

```suggestion
            labels_textprops = {
                'fontsize': mpl.rcParams['xtick.labelsize'],
                **cbook.normalize_kwargs(text_props or {}, Text)
            }
```